src/status-checks.ts

```diff
@@ -10,13 +38,57 @@ import { Router } from "./router"
  */
 export async function createStatusCheckComponent<Context extends object = {}>(components: {
   server: IHttpServerComponent<Context>
+  config: IConfigComponent
 }): Promise<IBaseComponent> {
   const { server } = components
 
   let mutStartOptions: IBaseComponent.ComponentStartOptions | undefined
 
   const routes = new Router()
 
+  const SUCCESSFUL_STATUS = 200
+  const FAILED_STATUS = 503
+  const MIME = "application/health+json; charset=utf-8"
+
+  async function getDetails(startup: boolean): Promise<StandardStatusResponse | null> {
+    if (!mutStartOptions) {
+      return null
+    }
+    const components: Record<string, IStatusCheckCapableComponent> = mutStartOptions.getComponents()
+
+    const probes: { name: string; promise: Promise<boolean> }[] = []
+
+    let functionName: "startupProbe" | "readynessProbe" = startup ? "startupProbe" : "readynessProbe"
+
+    for (let c in components) {
+      if (typeof components[c][functionName] == "function") {
+        probes.push({
+          name: c,
+          promise: new Promise((ok) => {
+            components[c][functionName]!()
+              .then(ok)
+              .catch(() => ok(false))
+          }),
+        })
+      }
+    }
+
+    const results = await Promise.all(probes.map(($) => $.promise))
+
+    const content: StandardStatusResponse = {
+      details: {},
+      status: results.some(($) => $ == false) ? "fail" : "pass",
```

Comment:
```suggestion
      status: results.some(($) => $ === false) ? "fail" : "pass",
```

In this case you want to make sure it's `false` and not any falsy value